extensions/bitwarden/src/components.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  showToast,
+  ToastStyle,
+  Form,
+  ActionPanel,
+  SubmitFormAction,
+  Detail,
+  CopyToClipboardAction,
+  getPreferenceValues,
+} from "@raycast/api";
+import { BitwardenApi } from "./api";
+
+export function TroubleshootingGuide(): JSX.Element {
+  showToast(ToastStyle.Failure, "Bitwarden CLI not found");
+  const { path } = getPreferenceValues();
+  const markdown = `# The Bitwarden CLI was not found
+## Please check that:
+
+1. The Bitwarden CLI is [correctly installed](https://bitwarden.com/help/article/cli/#download-and-install)
+1. The path of the installation matches the Bitwarden CLI Installation Path extension setting
+> Currently set to: \`${path}\`
+`;
+
+  return (
+    <Detail
+      markdown={markdown}
+      actions={
+        <ActionPanel>
+          <CopyToClipboardAction title={"Copy Homebrew Installation Command"} content="brew install bitwarden-cli" />
+        </ActionPanel>
+      }
+    />
+  );
+}
+
+export function UnlockForm(props: { setSessionToken: (session: string) => void, bitwardenApi: BitwardenApi }): JSX.Element {
+  async function onSubmit(values: { password: string }) {
+    try {
+      const toast = await showToast(ToastStyle.Animated, "Unlocking Vault...", "Please wait");
+      const sessionToken = await props.bitwardenApi.unlockVault(values.password)
+      toast.hide();
+
+      props.setSessionToken(sessionToken);
+    } catch (error) {
+      console.log(error);
+      showToast(ToastStyle.Failure, "Failed to unlock vault", "Invalid password");
```

Comment:
```suggestion
      showToast(ToastStyle.Failure, "Failed to unlock vault", "Invalid credentials");
```
It's a slightly more secure practice to avoid specifying exactly why a login failed. Telling a malicious user that the password is incorrect implies that the username _is_ correct, and could allow for brute force attacks by submitting tons of different passwords.

Reply:
Thanks, did not thought about that